src/parser/statement.js

```diff
@@ -771,8 +771,13 @@ pp.parseClassBody = function (node) {
 };
 
 pp.parseClassProperty = function (node) {
+  const noPluginMsg = "You can only use Class Properties when the 'classProperties' plugin is enabled.";
+  if (!node.typeAnnotation && !this.hasPlugin("classProperties")) {
+    this.raise(node.start, noPluginMsg);
+  }
+
   if (this.match(tt.eq)) {
-    if (!this.hasPlugin("classProperties")) this.unexpected();
+    if (!this.hasPlugin("classProperties")) this.raise(this.state.start, noPluginMsg);
```

Comment:
Should note I'm using `this.state.start` rather than `node.start` here, so we're pointing out the error where the `=` is, rather than what could be the start of a type annotation in the scenario that the code being parsed is:

```js
class Foo {
   bar: string = 'bizz';
}
```